pkg/postage/postagecontract/contract.go

```diff
@@ -199,7 +199,13 @@ func (c *postageContract) sendTransaction(ctx context.Context, callData []byte,
 	}
 
 	if receipt.Status == 0 {
-		return nil, transaction.ErrTransactionReverted
+		err := transaction.ErrTransactionReverted
+		if res, cErr := c.transactionService.Call(ctx, request); cErr == nil {
+			if reason, uErr := abi.UnpackRevert(res); uErr == nil {
+				err = fmt.Errorf("%w; reason: %s", err, reason)
```

Comment:
is the semicolon on purpose?

Reply:
> is the semicolon on purpose?

yes

Reply:
Don't we usually use a colon?

Reply:
I think this approach should suffice:
https://github.com/ethersphere/bee/blob/08bbf64b9513826e72f9f949c6cf0530fc4fdd26/pkg/node/node.go#L1287